_news/2019-10-07-3.1-release-notes/en.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Dodona 3.1"
+description: "Release notes Dodona 3.1"
+slug: "dodona-3.1"
+date: "2019-10-07"
+lang: en
+---
+
+The release of Dodona 3.1 makes Dodona more secure and faster.
+
+## Security
+
+Until now, exercises could change the entire layout of Dodona and execute _scripts_. These _scripts_ were able to do anything the user looking at the page could do on Dodona, e.g. submit solutions. For teachers this meant that these _scripts_ could be used to extract data about students. To be clear, this has never happened. However, since an ever increasing number of people can add exercises to Dodona, it became time to fix this.
+
+We fixed it by placing exercise descriptions in an _iframe_. This ensures that your browser will not allow the contents of the _iframe_ to do things outside of the _iframe_ (i.e. in the rest of Dodona). Additionally, we made sure that the contents of the feedback table of submitted solutions and the descriptions of courses and series are cleaned up before they are shown to users.
+
+## Speed
+
+For really large feedback tables it could take a while before the feedback table was shown after the submission was evaluated (sometimes up to 40 seconds!). By performing some smart tricks and where necessary showing the feedback in a simpler way we made sure that it never takes more than a second to load a feedback table.
```

Comment:
```suggestion
For really large feedback tables it could take a while before the feedback table was shown after the submission was evaluated (sometimes up to 40 seconds!). By performing some smart tricks and falling back to a simpler output format for large results, we managed to significantly reduce the loading time. Now, it should never takes more than a second to load a feedback table.
```